### PR TITLE
refactor: upgrade to jest v26

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

Jest v26 is out (https://jestjs.io/blog/2020/05/05/jest-26). The upgrade here should be relatively smooth due to the jsdom v16 adoption work earlier.

### CODE REVIEW

File: jest.test.config.js [hunk 1]
```diff
@@ -12,5 +12,5 @@ module.exports = {
     // Transpile also our local packages as they are only symlinked.
     'node_modules/(?!(@commercetools-[frontend|backend]+)/)',
   ],
-  testEnvironment: 'jest-environment-jsdom-sixteen',
+  testEnvironment: 'jsdom',
```
Comment (tdeekens) [inline jest.test.config.js:15]: jsdom "standard" now uses v16 of jsdom. The sixteen variant is not really needed anymore. I suggest however we keep the `jest-present-for-jsdom-sixteen` with the `jest-environment-jsdom-sixteen` as it otherwise would be breaking. We could there remove the `jest-environment-jsdom-sixteen` to `jsdom` but that could also be another PR at some point or just when it's removed.

**Comment by emmenko** [comment]
Finally ESM support: https://jestjs.io/blog/2020/05/05/jest-26#native-esm-support

**Comment by tdeekens** [comment]
> Finally ESM support: https://jestjs.io/blog/2020/05/05/jest-26#native-esm-support

Yup. It‘s super neat. Still, anything missing on this PR?

**Comment by tdeekens** [comment]
Thanks
